api/charmhub/data.go

```diff
@@ -4,143 +4,112 @@
+	if !ok {
+		logger.Errorf("unexpected: InfoEntity is not a bundle")
+		return Bundle{}
 	}
-	return result
-}
-
-func convertCategories(categories []params.Category) []Category {
-	result := make([]Category, len(categories))
-	for i, c := range categories {
-		result[i] = Category(c)
+	out := Bundle{
+		Charms: make([]BundleCharm, len(inB.Charms)),
 	}
-	return result
-}
-
-func convertChannelMap(cms []params.ChannelMap) []ChannelMap {
-	result := make([]ChannelMap, len(cms))
-	for i, cm := range cms {
-		result[i] = convertOneChannelMap(cm)
+	for i, c := range inB.Charms {
+		out.Charms[i] = BundleCharm(c)
 	}
-	return result
+	return out
 }
 
-func convertOneChannelMap(defaultMap params.ChannelMap) ChannelMap {
-	return ChannelMap{
-		Channel: Channel{
-			Name:       defaultMap.Channel.Name,
-			Platform:   Platform(defaultMap.Channel.Platform),
-			ReleasedAt: defaultMap.Channel.ReleasedAt,
-		},
-		Revision: Revision{
-			CreatedAt:    defaultMap.Revision.CreatedAt,
-			ConfigYAML:   defaultMap.Revision.ConfigYAML,
-			Download:     Download(defaultMap.Revision.Download),
-			MetadataYAML: defaultMap.Revision.MetadataYAML,
-			Revision:     defaultMap.Revision.Revision,
-			Version:      defaultMap.Revision.Version,
-			Platforms:    convertPlatforms(defaultMap.Revision.Platforms),
-		},
+func convertCharm(in interface{}) Charm {
+	inC, ok := in.(params.CharmHubCharm)
+	if !ok {
+		logger.Errorf("unexpected: InfoEntity is not a charm")
+		return Charm{}
+	}
+	return Charm{
+		Config:      params.FromCharmOptionMap(inC.Config),
+		Relations:   inC.Relations,
+		Subordinate: inC.Subordinate,
+		Tags:        inC.Tags,
+		UsedBy:      inC.UsedBy,
 	}
 }
 
-func convertPlatforms(platforms []params.Platform) []Platform {
-	result := make([]Platform, len(platforms))
-	for i, p := range platforms {
-		result[i] = Platform(p)
+func convertChannels(in map[string]params.Channel) map[string]Channel {
+	out := make(map[string]Channel, len(in))
+	for k, v := range in {
+		out[k] = Channel(v)
 	}
-	return result
+	return out
 }
 
 type InfoResponse struct {
-	Type           string       `json:"type"`
-	ID             string       `json:"id"`
-	Name           string       `json:"name"`
-	Charm          Charm        `json:"charm"`
-	ChannelMap     []ChannelMap `json:"channel-map"`
-	DefaultRelease ChannelMap   `json:"default-release,omitempty"`
-}
-
-type ChannelMap struct {
-	Channel  Channel  `json:"channel,omitempty"`
-	Revision Revision `json:"revision,omitempty"`
+	Type        string             `json:"type"`
+	ID          string             `json:"id"`
+	Name        string             `json:"name"`
+	Description string             `json:"description"`
+	Publisher   string             `json:"publisher"`
+	Summary     string             `json:"summary"`
+	Series      []string           `json:"series"`
+	StoreURL    string             `json:"store-url"`
+	Charm       Charm              `json:"charm"`
+	Bundle      Bundle             `json:"bundle"`
```

Comment:
I wonder if we should change this to the following? That way don't have a field we're not expected?

```suggestion
	Charm       *Charm              `json:"charm,omitempty"`
	Bundle      *Bundle             `json:"bundle,omitempty"`
```

Reply:
changed them all to pointers, it cause me to see that we had a mismatch of type values based on capitalization of first letter.  Need to type the Types!